fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java

```diff
@@ -779,13 +796,28 @@ private void replayCreateJob(SchemaChangeJobV2 replayedJob) throws MetaNotFoundE
 
             // set table state
             olapTable.setState(OlapTableState.SCHEMA_CHANGE);
+            Env.getCurrentEnv().getGroupCommitManager().setStatus(tableId, SchemaChangeStatus.BLOCK);
         } finally {
             olapTable.writeUnlock();
         }
 
         this.watershedTxnId = replayedJob.watershedTxnId;
         jobState = JobState.WAITING_TXN;
         LOG.info("replay pending schema change job: {}, table id: {}", jobId, tableId);
+        // wait wal done here
+        List<Long> aliveBeIds = Env.getCurrentSystemInfo().getAllBackendIds(true);
+        boolean walFinished = Env.getCurrentEnv().getGroupCommitManager()
+                .isPreviousWalFinished(tableId, watershedTxnId, aliveBeIds);
+        while (!walFinished) {
+            LOG.info("wai for wal queue size to be empty");
+            walFinished = Env.getCurrentEnv().getGroupCommitManager()
+                    .isPreviousWalFinished(tableId, watershedTxnId, aliveBeIds);
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException ie) {
+                LOG.info("schema change job sleep wait InterruptedException: ", ie);
```

Comment:
It's good practice to re-interrupt the current thread when it catches an InterruptedException.